### PR TITLE
11/10 hw1

### PR DESCRIPTION
**Pull request by container513**
## Feature/Bug Description
homework1 (file viewer)

## Solution
Implement the class diagram.

## Related Resource
[class diagram](https://app.diagrams.net/#G1O-CIwI_sFfggv0zRarxsAzGmwZ_uIKex)

## Result
image

### CODE REVIEW

File: 1110_hw1/Main.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    private static void parseCommandAdd(FileViewer viewer, String name, String[] elementNames) throws Exception {
+        for(String elementName: elementNames){
+            viewer.addElementForView(name, elementName);
+        }
+    }
+
+    private static void parseCommandNew(FileViewer viewer, String name, String text) {
+        viewer.attach(new TextView(name,text));
+    }
```
Comment (skychocowhite) [inline 1110_hw1/Main.java:53]: question: Do we need to handle duplicate viewer names?

File: 1110_hw1/FileViewer.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            element = new ThickBlackBorder();
+        }
+        else{
+            throw new Exception("The element not found");
+        }
+        views.stream()
+                .filter(view -> view.getName().equals(viewName))
+                .findFirst().orElseThrow(() -> new Exception("The TextViewName not found"))
+                .attach(element);
+    }
```
Comment (skychocowhite) [inline 1110_hw1/FileViewer.java:34]: **question**: Do we need to handle duplicate elements?

File: 1110_hw1/FileViewer.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        views.remove(view);
+    }
+    public void displayView(String viewName) throws Exception {
+        views.stream()
+                .filter(view -> view.getName().equals(viewName))
+                .findFirst().orElseThrow(() -> new Exception("The TextViewName not found"))
+                .display();
+    }
+
+    public void addElementForView(String viewName, String elementName) throws Exception {
```
Comment (shaoyuchu) [inline 1110_hw1/FileViewer.java:19]: nitpick: I personally prefer `addElementToView`

File: 1110_hw1/Main.java (hunk 1 above)
Comment (container513) [reply 1110_hw1/Main.java:53]: You are right. I forgot this condition.

File: 1110_hw1/FileViewer.java (hunk 2 above)
Comment (container513) [reply 1110_hw1/FileViewer.java:34]: You are right. I forgot this condition.

**Review by shaoyuchu — COMMENTED** [review]
suggestion: It would be better to make your PR title more precise. For example, `11/10 hw1`.

File: 1110_hw1/FileViewer.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import java.util.ArrayList;
+
+public class FileViewer {
+    private ArrayList<View> views = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (container513) [reply 1110_hw1/FileViewer.java]: OK

File: 1110_hw1/FileViewer.java (hunk 3 above)
Comment (container513) [reply 1110_hw1/FileViewer.java]: OK

**Review by skychocowhite — APPROVED** [review]
Well done!

**Review by shaoyuchu — APPROVED** [review]

**Review by Austin890720 — APPROVED** [review]
wonderful

**Pull request merged**